ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -196,22 +196,30 @@ private void analyzeNonStdOutSink(final Sink sink) {
       ));
     }
 
-    private KeyFormat buildKeyFormat() {
+    private KeyFormat buildKeyFormat(final Sink sink) {
       final Optional<KsqlWindowExpression> ksqlWindow = analysis.getWindowExpression()
           .map(WindowExpression::getKsqlWindowExpression);
 
+      final FormatInfo sourceKeyFormat = analysis
+          .getFrom()
+          .getDataSource()
+          .getKsqlTopic()
+          .getKeyFormat()
+          .getFormatInfo();
+
+      final FormatInfo keyFormat = sink.getProperties().getKeyFormatInfo()
+          .orElse(sourceKeyFormat);
+
+      // Todo: need to handle inheritable vs non-inheritable key properties
+
```

Comment:
This is the bit of code that handles getting the key format from the WITH properties for C*AS statements.